lib/calculate-class-breaks/normalize-classification-values.js

```diff
@@ -0,0 +1,123 @@
+const _ = require('lodash')
+
+function normalizeClassBreaks (features, classification) {
+  const { normType: type } = classification
+
+  if (type === 'field') {
+    return normalizeByField(features, classification)
+  }
+
+  if (type === 'log') {
+    return normalizeByLog(features, classification)
+  }
+
+  if (type === 'percent') {
+    return normalizeByPercent(features, classification)
+  }
+
+  throw new Error(`Normalization not supported: ${type}`)
+}
+
+function normalizeByField (features, { field: classificationField, normField: normalizationField }) {
+  if (!normalizationField) {
+    throw new Error('Normalization field is undefined')
+  }
+
+  if (!classificationField) {
+    throw new Error('Classification field is undefined')
+  }
+
+  const values = features.map(feature => {
+    if (shouldSkipFeatureForFieldNormalization({ feature, classificationField, normalizationField })) {
+      return
+    }
+
+    validateNormalizationValues({ feature, classificationField, normalizationField })
+
+    const valueToNormalizeBy = feature.properties[normalizationField] > 0 ? feature.properties[normalizationField] : 1
```

Comment:
It creates a misleading result. Instead, if a negative value is evil. the result can set to NaN or null, just to exclude the exception.

Reply:
Yes, I agree.  But this was just part of the legacy code and logic - see [here](https://github.com/koopjs/winnow/blob/master/lib/generateBreaks/normalizeClassBreaks.js#L39).  In this PR I was only trying to improve readability so that these types of problems are easier to identify.  Since I don't know what the original thinking was behind this logic, or how it is treated in ArcGIS server, I left it in place so not to release a breaking change.